nixos/modules/services/hardware/hddfancontrol.nix

```diff
@@ -52,21 +52,30 @@ in
             {
               options = {
                 disks = lib.mkOption {
-                  type = lib.types.listOf lib.types.path;
+                  type = lib.types.listOf lib.types.str;
                   default = [ ];
                   description = ''
                     Drive(s) to get temperature from
+
+                    Can also use command substitution to automatically grab all matching drives; such as all scsi (sas) drives
                   '';
-                  example = [ "/dev/sda" ];
+                  example = [
+                    "/dev/sda"
+                    "`find /dev/disk/by-id -name \"scsi*\" -and -not -name \"*-part*\" -printf \"%p \"`"
+                  ];
                 };
 
                 pwmPaths = lib.mkOption {
-                  type = lib.types.listOf lib.types.path;
+                  type = lib.types.listOf lib.types.str;
                   default = [ ];
                   description = ''
                     PWM filepath(s) to control fan speed (under /sys), followed by initial and fan-stop PWM values
+                    Can also use command substitution to ensure the correct hwmonX is selected on every boot
                   '';
-                  example = [ "/sys/class/hwmon/hwmon2/pwm1:30:10" ];
+                  example = [
+                    "/sys/class/hwmon/hwmon2/pwm1:30:10"
+                    "`echo /sys/devices/platform/nct6775.656/hwmon/hwmon[[:print:]]`/pwm4:80:20"
```

Comment:
Can `[[:print:]]` actually be used in shell globs like this?  I'm a bit lost figuring out how this works.